src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -99,7 +99,9 @@ def pileupSizeTask(self):
                 pileupSize = datasetSizes.get(pileupName, 0)
                 if pileupSize:
                     doc['pileupSize'] = pileupSize
-                self.mgr.updatePileup(doc, validate=False)
+                spec = {'pileupName': pileupName}
```

Comment:
Valentin, see my comment in the `updatePileupDocumentInDatabase` method. That way we don't need to pre-construct the spec object multiple times here before calling updatePileupDocumentInDatabase.
Please make sure to update all such occurrences.

Reply:
ok, done